packages/caliper-core/test/worker/rate-control/replayRate.js

```diff
@@ -0,0 +1,200 @@
+            sandbox.restore();
+        }
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it('should throw an error when pathTemplate is not provided', () => {
+        const msgContent = {
+            label: 'test',
+            rateControl: {
+                type: 'replay-rate',
+                opts: {}
+            },
+            workload: { module: 'module.js' },
+            testRound: 0,
+            txDuration: 250,
+            totalWorkers: 2
+        };
+        const testMessage = new TestMessage('test', [], msgContent);
+        const stubStatsCollector = sinon.createStubInstance(TransactionStatisticsCollector);
+
+        assert.throws(() => {
+            ReplayRate.createRateController(testMessage, stubStatsCollector, 0);
+        }, 'The path to load the recording from is undefined');
+    });
+
+    it('should throw an error when the trace file does not exist', () => {
+        const msgContent = {
+            label: 'test',
+            rateControl: {
+                type: 'replay-rate',
+                opts: {
+                    pathTemplate: './non-existent-file.txt'
+                }
+            },
+            workload: { module: 'module.js' },
+            testRound: 0,
+            txDuration: 250,
+            totalWorkers: 2
+        };
+        const testMessage = new TestMessage('test', [], msgContent);
+        const stubStatsCollector = sinon.createStubInstance(TransactionStatisticsCollector);
+
+        assert.throws(() => {
+            ReplayRate.createRateController(testMessage, stubStatsCollector, 0);
+        }, `Trace file does not exist: ${path.resolve('./non-existent-file.txt')}`);
+    });
+
+    it('should correctly import transaction timings from text format', () => {
+        const tempFilePath = path.join(__dirname, 'temp-replay-trace.txt');
+        const transactionTimings = [100, 200, 300, 400];
+        fs.writeFileSync(tempFilePath, transactionTimings.join('\n'), 'utf-8');
+
+        const msgContent = {
+            label: 'test',
+            rateControl: {
+                type: 'replay-rate',
+                opts: {
+                    pathTemplate: tempFilePath,
+                    inputFormat: 'TEXT'
+                }
+            },
+            workload: { module: 'module.js' },
+            testRound: 0,
+            txDuration: 250,
+            totalWorkers: 2
+        };
+        const testMessage = new TestMessage('test', [], msgContent);
+        const stubStatsCollector = sinon.createStubInstance(TransactionStatisticsCollector);
+        const replayRateController = ReplayRate.createRateController(testMessage, stubStatsCollector, 0);
+
+        assert.deepEqual(replayRateController.records, transactionTimings);
+
+        // Clean up the temporary file
+        fs.unlinkSync(tempFilePath);
+    });
+
+    it('should correctly apply rate control and log warning when running out of transaction timings', async () => {
```

Comment:
There are 2 different tests here, can we split them into 2 please. One where there are no log warnings and one where there are log warnings